src/Riok.Mapperly/Descriptors/DescriptorBuilder.cs

```diff
@@ -94,6 +96,47 @@ private void ConfigureMemberVisibility()
 #endif
     }
 
+    private bool IsInstanceWithStaticPartialMethod()
```

Comment:
This extracts the user method mappings an additional time. Is this really required? I think this could be simplified a lot with the following approach:
* Revert changes in `UserMethodMappingExtractor`
* In `Riok.Mapperly.Descriptors.DescriptorBuilder.ExtractUserMappings`:
    * Use a boolean to track whether all mappings should be static, initialize it to true if the mapper class itself is static
    * Store the first non-static user mapping method symbol
    * Simplified, it could look like this:
    ```c#
        var allShouldBeStatic = _mapperDescriptor.Symbol.IsStatic;
        IMethodSymbol? firstNonStaticUserMapping = null;

        foreach (...)
        {
            if (userMapping.Method.IsStatic)
            {
                allShouldBeStatic = true;
            }
            else if (firstNonStaticUserMapping == null)
            {
                firstNonStaticUserMapping = userMapping.Method;
            }

            ...
        }
        ...
     ```
     * after the extraction of all user mappings is done report the diagnostic on `firstNonStaticUserMapping` if `firstNonStaticUserMapping` is not null and `allShouldBeStatic` is true.